modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/chapters/ChaptersViewModel.kt

```diff
@@ -0,0 +1,109 @@
+package au.com.shiftyjelly.pocketcasts.player.view.chapters
+
+import androidx.compose.ui.graphics.Color
+import androidx.lifecycle.ViewModel
+import au.com.shiftyjelly.pocketcasts.models.entity.Podcast
+import au.com.shiftyjelly.pocketcasts.models.to.Chapter
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
+import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackState
+import au.com.shiftyjelly.pocketcasts.repositories.playback.UpNextQueue
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.EpisodeManager
+import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
+import au.com.shiftyjelly.pocketcasts.ui.theme.Theme
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.BackpressureStrategy
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+@HiltViewModel
+class ChaptersViewModel
+@Inject constructor(
+    episodeManager: EpisodeManager,
+    podcastManager: PodcastManager,
+    private val playbackManager: PlaybackManager,
+    private val theme: Theme
+) : ViewModel(), CoroutineScope {
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    data class UiState(
+        val chapters: List<ChapterState> = emptyList(),
+        val backgroundColor: Color
+    )
+
+    data class ChapterState(
+        val chapter: Chapter,
+        val isPlayed: Boolean,
+        val isPlaying: Boolean,
```

Comment:
I did try using a playing status for this but it seemed to add more code so I ended up removing it. I'm not sure if that was the right decision. 🤔 
```
    sealed class ChapterPlayingStatus {
        object Played: ChapterPlayingStatus()
        data class Playing(val progress: Float): ChapterPlayingStatus()
        object NotPlayed: ChapterPlayingStatus()
    }
```

Reply:
I like how you changed the ChapterState instead I added a new ChapterPlayingStatus sealed class. I'm going to steal your diff. Thanks!